spec/models/petition_spec.rb

```diff
@@ -474,117 +468,173 @@
     end
   end
 
-  describe "can_be_signed?" do
-    def petition(state = Petition::OPEN_STATE)
-      FactoryGirl.build(:petition, :state => state)
+  describe "#can_be_signed?" do
+    context "when the petition is in the open state" do
+      let(:petition) { FactoryGirl.build(:petition, state: Petition::OPEN_STATE) }
+
+      it "is true" do
+        expect(petition.can_be_signed?).to be_truthy
+      end
     end
 
-    it "is true if and only if the petition is OPEN and the closed_at date is in the future" do
-      petition = FactoryGirl.build(:open_petition, :closed_at => 1.year.from_now)
-      expect(petition.can_be_signed?).to be_truthy
+    (Petition::STATES - [Petition::OPEN_STATE]).each do |state|
+      context "when the petition is in the #{state} state" do
+        let(:petition) { FactoryGirl.build(:petition, state: state) }
+
+        it "is false" do
+          expect(petition.can_be_signed?).to be_falsey
+        end
+      end
     end
+  end
 
-    it "is false if the petition is OPEN and the closed_at date is in the past" do
-      petition = FactoryGirl.build(:open_petition, :closed_at => 2.minutes.ago)
-      expect(petition.can_be_signed?).to be_falsey
+  describe "open?" do
+    context "when the state is open" do
+      let(:petition) { FactoryGirl.build(:petition, state: Petition::OPEN_STATE) }
+
+      it "returns true" do
+        expect(petition.open?).to be_truthy
+      end
     end
 
-    it "is false otherwise" do
-      expect(petition(Petition::PENDING_STATE).can_be_signed?).to be_falsey
-      expect(petition(Petition::HIDDEN_STATE).can_be_signed?).to be_falsey
-      expect(petition(Petition::REJECTED_STATE).can_be_signed?).to be_falsey
-      expect(petition(Petition::VALIDATED_STATE).can_be_signed?).to be_falsey
-      expect(petition(Petition::SPONSORED_STATE).can_be_signed?).to be_falsey
+    context "for other states" do
+      (Petition::STATES - [Petition::OPEN_STATE]).each do |state|
+        let(:petition) { FactoryGirl.build(:petition, state: state) }
+
+        it "is not open when state is #{state}" do
+          expect(petition.open?).to be_falsey
+        end
+      end
     end
   end
 
-  describe "open?" do
-    it "is open when state is open" do
-      expect(FactoryGirl.build(:petition, :state => Petition::OPEN_STATE).open?).to  be_truthy
+  describe "#closed?" do
+    context "when the closed_at has passed but the state is still open" do
+      let(:petition) { FactoryGirl.build(:petition, state: Petition::CLOSED_STATE, closed_at: 3.days.ago) }
+
+      it "return true" do
+        expect(petition.closed?).to be_truthy
+      end
+    end
+
+    context "when the closed_at has not passed but the state is closed" do
+      let(:petition) { FactoryGirl.build(:petition, state: Petition::CLOSED_STATE, closed_at: 3.days.from_now) }
+
+      it "return true" do
+        expect(petition.closed?).to be_truthy
+      end
```

Comment:
Can we just collapse these down to "is closed when state is closed"? like the others?  I admit that I find the title of `"when the closed_at has passed but the state is still open"` a bit confusing (compared to the implementation which uses `CLOSED_STATE`, not `OPEN_STATE`.